tests/network/mod.rs

```diff
@@ -99,21 +105,56 @@ impl MessageScheduler {
     }
 }
 
-pub type MessageWithSender<D> = (
-    <D as DistAlgorithm>::NodeUid,
-    TargetedMessage<<D as DistAlgorithm>::Message, <D as DistAlgorithm>::NodeUid>,
-);
+/// A message combined with a sender
+pub struct MessageWithSender<D: DistAlgorithm> {
+    /// The sender of the message
+    pub sender: <D as DistAlgorithm>::NodeUid,
+    /// The targeted message (recipient and message body)
+    pub tm: TargetedMessage<<D as DistAlgorithm>::Message, <D as DistAlgorithm>::NodeUid>,
+}
+
+// the Debug implementation cannot be derived automatically, possibly due to a compiler bug
+impl<D: DistAlgorithm> fmt::Debug for MessageWithSender<D> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "<MessageWithSender[from={:?} to={:?}]{:?}>",
```

Comment:
I considered it - the issue is that if MessageWithSender is changed later (e.g. by adding a field), it might be easier to see why the debug output has not been affected.

On second thought I agree though. Fixed in d3f30fd